apps/meteor/tests/unit/app/lib/server/lib/notifyListener.spec.ts

```diff
@@ -0,0 +1,215 @@
+import type { IMessage } from '@rocket.chat/core-typings';
+import { expect } from 'chai';
+import proxyquire from 'proxyquire';
+import sinon from 'sinon';
+
+describe('Message Broadcast Tests', () => {
+    let getSettingValueByIdStub: sinon.SinonStub;
+    let usersFindOneStub: sinon.SinonStub;
+    let messagesFindOneStub: sinon.SinonStub;
+    let broadcastStub: sinon.SinonStub;
+    let getMessageToBroadcast: any;
+    let notifyOnMessageChange: any;
+    let memStub: sinon.SinonStub;
+
+    const sampleMessage: IMessage = {
+        _id: '123',
+        rid: 'room1',
+        msg: 'Hello',
+        ts: new Date(),
+        u: { _id: 'user1', username: 'user1', name: 'Real User' },
+        mentions: [],
+        t: 'user-muted',
+        _updatedAt: new Date(),
+    };
+
+    const modelsStubs = () => ({
+        Messages: {
+            findOneById: messagesFindOneStub,
+        },
+        Users: {
+            findOne: usersFindOneStub,
+        },
+        Settings: {
+            getValueById: getSettingValueByIdStub,
+        },
+    });
+
+    const coreStubs = (dbWatchersDisabled: boolean) => ({
+        api: {
+            broadcast: broadcastStub,
+        },
+        dbWatchersDisabled,
+    });
+
+    beforeEach(() => {
+        getSettingValueByIdStub = sinon.stub();
+        usersFindOneStub = sinon.stub();
+        messagesFindOneStub = sinon.stub();
+        broadcastStub = sinon.stub();
+        memStub = sinon.stub().callsFake((fn: any) => fn);
+
+        const proxyMock = proxyquire.noCallThru().load('../../../../../../app/lib/server/lib/notifyListener', {
+            '@rocket.chat/models': modelsStubs(),
+            '@rocket.chat/core-services': coreStubs(false),
+            'mem': memStub,
+        });
+
+        getMessageToBroadcast = proxyMock.getMessageToBroadcast;
+        notifyOnMessageChange = proxyMock.notifyOnMessageChange;
+    });
+
+    afterEach(() => {
+        sinon.reset();
+    });
+
+    describe('getMessageToBroadcast', () => {
+        let originalEnv: NodeJS.ProcessEnv;
+
+        beforeEach(() => {
+            originalEnv = { ...process.env };
+            sinon.resetHistory();
+        });
+
+        afterEach(() => {
+            process.env = originalEnv;
+        });
+
+        const testCases = [
+            {
+                description: 'should return undefined if message is hidden or imported',
```

Comment:
Can we have one case for each message? This one is the hidden one, can we create a specific one for the imported one?